### PR TITLE
Added tests for #148

### PR DESCRIPTION
**Pull request by Den1552**
## Summary 

- Tests behaviour of fixed issue #148. 

## Added

- Added new test file (`vcast_build_env_failure_different_paths.test`) to test multiple builds with vc23 and vc24.

## Changed

- Modified `wdio.conf.ts` to build multiple envs for `vcast_build_env_failure_different_paths.test`.
- Modified `specs_config.ts` and created new specs group.

### CODE REVIEW

File: tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    const activityBar = workbench.getActivityBar();
+    const testingView = await activityBar.getViewControl("Testing");
+    await testingView?.openView();
+
+    // Open Testing
+    const vcastTestingViewContent: ViewContent =
+      await getViewContent("Testing");
+    let subprogram: TreeItem;
+
+    // Iterate through Testing and try to expand required Envs (1 & 3)
```
Comment (Zbigor) [inline tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts:92]: Maybe you could write some `expect` assertions for this test (and other similar ones)

File: tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    vcastTestingViewContent = await getViewContent("Testing");
+    for (const vcastTestingViewSection of await vcastTestingViewContent.getSections()) {
+      if (!(await vcastTestingViewSection.isExpanded())) {
+        await vcastTestingViewSection.expand();
+      }
+
+      // Expand envs if found
+      for (const vcastTestingViewContentSection of await vcastTestingViewContent.getSections()) {
+        await vcastTestingViewContentSection.expand();
+
```
Comment (Zbigor) [inline tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts:193]: Can you try to reduce code repetition, in particular for expanding multiple environments. Maybe you can write a helper function for this or extend an existing function

File: tests/internal/e2e/test/wdio.conf.ts [hunk 3]
```diff
@@ -309,207 +315,317 @@ export const config: Options.Testrunner = {
...
+      const createCFG = `cd ${workspacePath} && ${process.env.VECTORCAST_DIR}/clicast -lc template GNU_CPP17_X`;
+      await promisifiedExec(createCFG);
+
+      // Build environments based on the iteration index
+      for (let i = 1; i <= totalEnvCount; i++) {
+        // ENV_0i
+        const envName = `ENV_${i.toString().padStart(2, "0")}`;
+        const envFile = path.join(workspacePath, `${envName}.env`);
+
+        // Switch VectorCAST version based on iteration (build 1,3 --> release 23, 2,4 --> release 24)
```
Comment (Zbigor) [inline tests/internal/e2e/test/wdio.conf.ts:605]: maybe adding a VectorCAST version indicator to environment names could make this less confusing and remove the need for the comment, though it's clear what's going on

File: tests/internal/e2e/test/wdio.conf.ts [hunk 4]
```diff
@@ -750,3 +866,6 @@ ENVIRO.END
   // onReload: function(oldSessionId, newSessionId) {
   // }
 };
+function copyDirectory(sourceFile: string, destinationFile: string) {
```
Comment (Zbigor) [inline tests/internal/e2e/test/wdio.conf.ts:869]: Should we remove this if it's neither used nor implemented

**Review by Zbigor — CHANGES_REQUESTED** [review]
Looks alright so far.The main change to be made is to try to reduce code duplication in the new E2E tests if possible. Other comments are mostly nits

File: tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts (hunk 2 above)
Comment (Den1552) [reply tests/internal/e2e/test/specs/vcast_build_env_failure_different_paths.test.ts]: Reduced code repetiotion in 7c690c6fda33b48239ce5fb7aeb15024a3df53d6.

File: tests/internal/e2e/test/wdio.conf.ts (hunk 3 above)
Comment (Den1552) [reply tests/internal/e2e/test/wdio.conf.ts:608]: Added release prefix to envs in 7c690c6fda33b48239ce5fb7aeb15024a3df53d6.

File: tests/internal/e2e/test/wdio.conf.ts (hunk 4 above)
Comment (Den1552) [reply tests/internal/e2e/test/wdio.conf.ts]: Removed in 7c690c6fda33b48239ce5fb7aeb15024a3df53d6.

**Review by Zbigor — APPROVED** [review]

**Pull request merged**